src/main.ts

```diff
@@ -13,15 +14,27 @@ config.load();
 const container = new Container();
 
 let store: PersistentStorage<number|string>;
+let DBStore: any;
+
+switch(config.dbMode) {
+    case ConfigManager.PG:
+        DBStore = PostgresStore;
+        break;
+    case ConfigManager.MONGO:
+        DBStore = MongodbStore;
+        break;
+    default:
+        throw new Error('DB_MODE is not support yet');
+}
 
 switch (config.mode) {
     case ConfigManager.DMHY:
-        container.bind<PersistentStorage<number>>(TYPES.PersistentStorage).to(PostgresStore).inSingletonScope();
+        container.bind<PersistentStorage<number>>(TYPES.PersistentStorage).to(DBStore).inSingletonScope();
```

Comment:
This could move to the switch(config.dbMode) code block

Reply:
In `DMHY` mode we need bind PersistentStorage with `number`, in `BANGUMI_MOE` mode
we need bind `PersistentStorage` wtth `string`. So, we need add if else to resolve it ?

Reply:
Oh, I got it. then your solution is ok